pkgs/applications/editors/emacs/make-emacs.nix

```diff
@@ -96,10 +96,11 @@
   withX ? !(stdenv.hostPlatform.isDarwin || noGui || withPgtk),
   withXinput2 ? withX,
   withXwidgets ?
-    !stdenv.hostPlatform.isDarwin
-    && !noGui
-    && (withGTK3 || withPgtk)
-    && (lib.versionOlder version "30"), # XXX: upstream bug 66068 precludes newer versions of webkit2gtk (https://lists.gnu.org/archive/html/bug-gnu-emacs/2024-09/msg00695.html)
+    !noGui
+    && (withGTK3 || withPgtk || withNS || variant == "macport")
+    && (stdenv.hostPlatform.isDarwin || lib.versionOlder version "30"),
```

Comment:
@alanpearce Why are we switching this on by default on darwin? If we just want to enable it for the gui versions then the previous condition on line 100 should be ok.

Reply:
@tani thanks for clarifying. Looks weird though, but I wouldn't have an alternative to achieve the same.